public/controllers/management/components/management/ruleset/ruleset-editor.js

```diff
@@ -88,6 +90,12 @@ class WzRulesetEditor extends Component {
     this._isMounted = true;
   }
 
+  async componentDidUpdate(prevProps) {
+    if (this.props.restoreFileContent) {
+      this.setState({content: this.state.initContent, restoreFileContent: false})      
+    }
+  }
+
```

Comment:
I think it is not necessary. You could remove the `restoreFileContent` state because it isn't used. The `content` state is reset to the same value of `initContent` when there is an error validation in the manager.